pkg/endpoint/types.go

```diff
@@ -36,12 +36,13 @@ type Provider interface {
 
 // Endpoint is a tuple of IP and Port representing an instance of a service
 type Endpoint struct {
-	net.IP `json:"ip"`
-	Port   `json:"port"`
+	net.IP   `json:"ip"`
+	Port     `json:"port"`
+	ProxyUID string `json:"proxy_uid"`
 }
 
 func (ep Endpoint) String() string {
-	return fmt.Sprintf("(ip=%s, port=%d)", ep.IP, ep.Port)
+	return fmt.Sprintf("(ip=%s, port=%d, proxy UID=%s)", ep.IP, ep.Port, ep.ProxyUID)
```

Comment:
This is a great point @nojnhuh. My only concern with the ProxyUID being encoded in the Endpoints struct isif we would be able to derive this mapping from an endpoint -> ProxyUID with other endpoint providers we add in the future. I think since we are not supporting additional providers at the moment, this encoding is not much of a concern now. Just something to think about.